doc/proposal/thrift-parameter-validation-proposal.md

```diff
@@ -0,0 +1,182 @@
+| le           | less equal (`<=`)                | i8, i16, i32, i64, f64               | same with value        | -                  |
+| gt           | greater than (`>`)               | i8, i16, i32, i64, f64               | same with value        | -                  |
+| ge           | greater equal (`>=`)             | i8, i16, i32, i64, f64               | same with value        | -                  |
+| in           | within given container           | i8, i16, i32, i64, f64, string, bool | same with value        | -                  |
+| not_in       | not within given container       | i8, i16, i32, i64, f64, string, bool | same with value        | -                  |
+| elem         | field's element constraint       | any                                  | list, set              | support            |
+| key          | field's element key constraint   | any                                  | map                    | support            |
+| value        | field's element value constraint | any                                  | map                    | support            |
+| min_size     | minimal length                   | i8, i16, i32, i64                    | string, list, set, map | -                  |
+| max_size     | maximal length                   | i8, i16, i32, i64                    | string, list, set, map | -                  |
+| prefix       | field prefix must be             | string                               | string                 | -                  |
+| suffix       | suffix must be                   | string                               | string                 | -                  |
+| contain      | must contain                     | string                               | string                 | -                  |
+| not_contain  | must not contain                 | string                               | string                 | -                  |
+
+- Secodary validator is a successive validator, usually used at container-type field. See below Set/List/Map examples.
+
+#### 3.2.2 IDL example
+- Number
+```
+struct NumericDemo{
+    1: double Value (validator.gte = "1000.1", validator.lte = "10000.1")
+    2: i8 Type (validator.in = "1", validator.in = "2", validator.in = "4")
+    3: i64 MagicNumber (validator.const = "0x5f3759df") 
+}
+```
+- String/Binary
+``` 
+struct StringDemo{
+    1: string Uninitialized (vt.const = "abc")
+    2: string Name (vt.min_size = "6", vt.max_size = "12")
+    3: string SomeStuffs (vt.pattern = "[0-9A-Za-z]+")
+    4: string DebugInfo (vt.prefix = "[Debug]")
+    5: string PanicInfo (vt.contains = "panic")
+    6: string Editor (vt.in = "vscode", vt.in = "vim", vt.in = "goland")
+}
+```
+- Bool
+```
+struct BoolDemo {
+    1: bool AMD (vt.const = "true")
+}
+```
+- Enum
+```
+struct EnumDemo {
+    1: Type AddressType (vt.const = "String")
```

Comment:
with the above example of `vt.in`, I don't see how `vt.const` is different from `vt.in`? can you explain the differences between them?

Reply:
"vt.const" and "vt.in" both are used to limit the range of values. The difference is that "vt.const" only specifies one value, "vt.in" multiple values can be specified